### ISSUE TITLE
DeleteHandlers (former DeleteImplicits) doesn't take into account size parameter

### ISSUE THREAD

**Issue by GMadorell** [sksamuel/elastic4s#1375]
Hi!

I've been playing with elastic4s lately, trying to delete some documents by query.

I was trying to limit the amount of cleaned documents, using the size field as such:

```scala
deleteIn(index / `type`).by(query).size(limit).abortOnVersionConflict(true)
```

But then I realized this parameter is not marshalled into the request body or as a url parameter.

Example of the code that should do this but doesn't.

Version: 5.6.6

com.sksamuel.elastic4s.http.delete.DeleteImplicits

```scala
implicit object DeleteByQueryExecutable extends HttpExecutable[DeleteByQueryDefinition, DeleteByQueryResponse] {

    override def execute(client: RestClient, request: DeleteByQueryDefinition): Future[DeleteByQueryResponse] = {

      // ...
request.requestsPerSecond.map(_.toString).foreach(params.put("requests_per_second", _))
      request.timeout.map(_.toMillis + "ms").foreach(params.put("timeout", _))
      request.scrollSize.map(_.toString).foreach(params.put("scroll_size", _))
      // HERE IT SHOULD BE ADDING THE SIZE PARAMETER
      
      // ...
    }
  }
```

Is that right, or am I missing something? If I have some time in the near future I might add a PR for this but atm I want to make sure this is really a bug so I don't spend time for nothing.

Thanks!

**Comment by GMadorell** [comment]
Hey, thanks for the fast reply!

Yeah, we need to delete a specific maximum amount of documents at once, which the scroll_size doesn't do. We need the 'size' parameter, not the 'scroll_size' parameter. The 'scroll_size' parameter works fine.

I have fixed it providing a custom executor which has the 'size' parameter on the parameter map. It would be nice to have this at least in the docs (if it's easy to change), so that if anyone has the same use case need as me, at least doesn't lose a lot of time.

**Comment by sksamuel** [comment]
I cannot see where size on a delete request is supported in the docs. Can you link me?
Or are you setting it on the query object.

**Comment by GMadorell** [comment]
Yeah, I'm setting it on the query object.

This is the code I'm running atm:

```scala
deleteIn(index / `type`).by(query).size(limit).abortOnVersionConflict(true)
```

**Comment by sksamuel** [comment]
I'm not sure .size makes any difference here, as they don't mention it in:
https://www.elastic.co/guide/en/elasticsearch/reference/5.6/docs-delete-by-query.html

**Comment by GMadorell** [comment]
Well, I manually tried adding the size=XX as a url parameters and it effectively limits the amount of deleted items, I'm running that code in production atm and it was crashing without the change to add it as a url parameter. I saw it somewhere on the docs as well but basically it works by limiting the query on the delete-by-query use case.

**Comment by sksamuel** [comment]
Ok, so this should work,

`deleteIn(index / `type`).by(query.size(limit)).abortOnVersionConflict(true)`

**Comment by GMadorell** [comment]
I'll make sure to test that, thanks! I'll tell you my results ;)

Should this work, though?

```scala
deleteIn(index / `type`).by(query).size(limit).abortOnVersionConflict(true)
```

If it shouldn't maybe it would be a good idea to remove the size parameter in the query dto. If it should, I'll try to make a PR when I have time.

**Comment by sksamuel** [comment]
Well if it does work I'm surprised because the elastic docs don't mention anything about it. It it doesn't work then it's a bug, perhaps a leftover from version 2.
The query.size seems probably the right way.

**Comment by GMadorell** [comment]
Allright, I can confirm that the following doesn't compile:

```scala
deleteIn(index / `type`).by(query.size(limit)).abortOnVersionConflict(true)
```

**Comment by sksamuel** [comment]
I've fixed this in 7.0.1, so you can do `deleteByQuery("index", query).size(x)` and the parameter is correctly picked up.
